app/test/testmerginapi.cpp

```diff
@@ -114,6 +116,83 @@ void TestMerginApi::testDownloadProject()
   passedTests++;
 }
 
+void TestMerginApi::testCancelDownlaodProject()
+{
+  qDebug() << "TestMerginApi::testCancelDownlaodProject START";
+  runTests++;
+  QString projectName = TestMerginApi::TEST_PROJECT_NAME_DOWNLOAD;
+
+  // create a project
+  QSignalSpy spy( mApi, SIGNAL( projectCreated( QString ) ) );
+  mApi->createProject( mUsername, projectName );
+  QVERIFY( spy.wait( SHORT_REPLY ) );
+  QCOMPARE( spy.count(), 1 );
+
+  // Copy data
+  QString source = mTestData + "/" + TestMerginApi::TEST_PROJECT_NAME + "/";
+  QString projectDir = mProjectModel->dataDir() + projectName + "/";
+  qDebug() << source << projectDir;
+  InputUtils::cpDir( source, projectDir );
+
+  // Upload data
+  MerginProject project;
+  std::shared_ptr<MerginProject> p = std::make_shared<MerginProject>( project );
+  p->name = projectName;
+  p->projectNamespace = mUsername;
+  p->projectDir = projectDir;
+  mApi->addProject( p );
+
+  QSignalSpy spy3( mApi, SIGNAL( syncProjectFinished( QString, QString, bool ) ) );
+  mApi->uploadProject( mUsername, projectName );
+  QVERIFY( spy3.wait( LONG_REPLY ) );
+  QCOMPARE( spy3.count(), 1 );
+  QList<QVariant> arguments = spy3.takeFirst();
+  QVERIFY( arguments.at( 2 ).toBool() );
+
+  // Remove the whole project
+  QDir( projectDir ).removeRecursively();
+  QFileInfo info( projectDir );
+  QDir dir( projectDir );
+  QCOMPARE( info.size(), 0 );
+  QVERIFY( dir.isEmpty() );
+  mApi->clearProject( p );
```

Comment:
It would be good to have a method in MerginApi to remove the whole project and so ProjectModel::deleteProject() would also use that API call. This would avoid the need to have projectDeletedOnPath() signal + slot in MerginApi. Not needed to address now, though...